src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java

```diff
@@ -692,93 +690,124 @@ public void scan(HttpMessage msg, String param, String origParamValue) {
 					if (this.debugEnabled) {
 						log.debug("The parameter value [" + origParamValue + "] is of type Integer");
 					}
+					// This check is implemented using two variant PLUS(+) and MULT(*)
+					// PLUS variant check the param value "3-2" gives same result and param value "4-2" gives different result if original param value is 1
+					// PLUS variant check the param value "2/2" gives same result and param value "4/2" gives different result if original param value is 1
+					String check4Variant []= {"plus", "mult"};
 
-					//get a value 2 sizes bigger
-					int paramPlusTwo = paramAsInt + 2;
-					String modifiedParamValue = String.valueOf(paramPlusTwo) + "-2";
-
-					//and prepare a request to set the parameter value to a string value like "3-2", if the original parameter value was "1"
-					//those of you still paying attention will note that if handled as expressions (such as by a database), these represent the same value.
-					HttpMessage msg4 = getNewMsg();
-					setParameter(msg4, param, modifiedParamValue);
-
-					try {
-						sendAndReceive(msg4, false); //do not follow redirects
-					} catch (SocketException ex) {
-						if (log.isDebugEnabled()) log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
-								" when accessing: " + msg4.getRequestHeader().getURI().toString());
-						return; //Something went wrong, no point continuing
-					}
-					countExpressionBasedRequests++;
-
-					String modifiedExpressionOutputUnstripped = msg4.getResponseBody().toString();
-					String modifiedExpressionOutputStripped = this.stripOff(modifiedExpressionOutputUnstripped, modifiedParamValue);
-
-					//set up two little arrays to ease the work of checking the unstripped output, and then the stripped output
-					String normalBodyOutput[] = {mResBodyNormalUnstripped, mResBodyNormalStripped};
-					String expressionBodyOutput[] = {modifiedExpressionOutputUnstripped, modifiedExpressionOutputStripped};
-					boolean strippedOutput[] = {false, true};
+					for (int booleancheck4VariantIndex = 0; booleancheck4VariantIndex<2 && !sqlInjectionFoundForUrl && countExpressionBasedRequests < doExpressionMaxRequests; booleancheck4VariantIndex++) {
```

Comment:
The c in check should be capitalized `booleanCheck4VariantIndex `.  (Is the leading "boolean" necessary? If so shouldn't it actually be "expression"?)
Also probably best to check against `check4Variant.length` instead of statically `2` (make maintenance more friendly in the future if another variant is added).